persistent/Database/Persist/Class/PersistEntity.hs

```diff
@@ -221,6 +224,8 @@ deriving instance (Eq (Key record), Eq record) => Eq (Entity record)
 deriving instance (Ord (Key record), Ord record) => Ord (Entity record)
 deriving instance (Show (Key record), Show record) => Show (Entity record)
 deriving instance (Read (Key record), Read record) => Read (Entity record)
+deriving instance (Data (Key record), Data record) => Data (Entity record)
+deriving instance (Typeable (Key record), Typeable record) => Typeable (Entity record)
```

Comment:
This shouldn't be necessary - `Typeable` is automagically derived for all types as of relatively recent GHCs.